### DISCUSSION TITLE
failing to setup plugin

### DISCUSSION THREAD

**Discussion by o0Noise0o** [LazyVim/LazyVim#4154] [Q&A]
Hello all, i would like to add the following plugin to my up to date lazyvim config: https://github.com/christoomey/vim-tmux-navigator I am new into this field and trying to get a hang of it but i am stuck. Hopefully u know what i am doing wrong.

In the init.lua file i have set the following first:

defaults = { -- By default, only LazyVim plugins will be lazy-loaded. Your custom plugins will load during startup. -- If you know what you're doing, you can set this to true to have all your custom plugins lazy-loaded by default. lazy = true, -- It's recommended to leave version=false for now, since a lot the plugin that support versioning, -- have outdated releases, which may break your Neovim install. version = "", -- always use the latest git commit -- version = "", -- try installing the latest stable version for plugins that support semver },

To enable the functionality of using .lua files in the plugins folder. IN the plugin folder i made an new file called tmuxnavigator.lua in this .lua file i placed ( according to lazy.vim installation in the github repo: { "christoomey/vim-tmux-navigator", cmd = { "TmuxNavigateLeft", "TmuxNavigateDown", "TmuxNavigateUp", "TmuxNavigateRight", "TmuxNavigatePrevious", }, keys = { { "<c-h>", "<cmd><C-U>TmuxNavigateLeft<cr>" }, { "<c-j>", "<cmd><C-U>TmuxNavigateDown<cr>" }, { "<c-k>", "<cmd><C-U>TmuxNavigateUp<cr>" }, { "<c-l>", "<cmd><C-U>TmuxNavigateRight<cr>" }, { "<c-\>", "<cmd><C-U>TmuxNavigatePrevious<cr>" }, }, }

So now i expected that by starting nvim i would be able to install using lazy.vim But when i open nvim i get an error:

Failed to load plugins.tmuxnavigator

...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/plugin.lua:156: attempt to call a nil value

stacktrace:
~/.config/nvim/lua/config/lazy.lua:17

~/.config/nvim/init.lua:2

and at this point i am completely stuck.

**Comment by folke** [comment]
every plugin lua file should return a table of plugin specs.
So wrap your code in `return { your_code }`

**Reply by o0Noise0o** [reply]
Thank u Folke, everything is working as intended.

**Comment by o0Noise0o** [comment]
Closed because Folke saw what i did wrong. See comment
